### PR TITLE
feat: credential branding

### PR DESCRIPTION
**Pull request by amanji**
# Summary of Changes

This PR introduces a feature for customizing credential look (branding) based on a template overlay that can be specified as a configuration.

The overlay provides the ability to:
* Style the background colour
* Apply a background image to credentials
* Style the header background and text colour
* Add a branding logo to the header
* Style the footer text colour

Other items:
* Revocation and credential objects no longer need to be passed into the credential card component.
* If an overlay is not supplied, credentials will automatically apply a background colour and light/dark text colour.
* Credentials will automatically apply styling upon revocation
* Refactored test contexts into separate files and updated test/snapshots
* Utility functions removed

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [ ] Updated documentation as needed for changed code and new or modified features;
- [ ] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: core/App/components/misc/CredentialCard.tsx [hunk 1]
```diff
@@ -1,93 +1,254 @@
...
+import { getCredentialConnectionLabel, hashCode, hashToRGBA, parsedCredDefName } from '../../utils/helpers'
+import { luminanceForHexColour } from '../../utils/luminance'
+import { credentialSchema } from '../../utils/schema'
 import { testIdWithKey } from '../../utils/testable'
 
-import AvatarView from './AvatarView'
-
 interface CredentialCardProps {
   credential: CredentialExchangeRecord
-  revoked?: boolean
```
Comment (wadeking98) [inline core/App/components/misc/CredentialCard.tsx:18]: are we getting rid of support for revoked credentials

File: core/App/screens/ProofRequest.tsx [hunk 2]
```diff
@@ -124,7 +124,9 @@ const ProofRequest: React.FC<ProofRequestProps> = ({ navigation, route }) => {
     setLoading(true)
     const retrieveCredentialsForProof = async (proof: ProofRecord) => {
       try {
-        const credentials = await agent.proofs.getRequestedCredentialsForProofRequest(proof.id)
+        const credentials = await agent.proofs.getRequestedCredentialsForProofRequest(proof.id, {
```
Comment (wadeking98) [inline core/App/screens/ProofRequest.tsx:127]: should proof requests with revoked credentials be in a separate PR?

File: core/App/screens/ListCredentials.tsx [hunk 3]
```diff
@@ -28,21 +25,22 @@ const ListCredentials: React.FC = () => {
...
-        >
-          <CredentialCard
-            credential={credential}
-            revoked={revoked.has(credential.id)}
-            onPress={() => navigation.navigate(Screens.CredentialDetails, { credentialId: credential.id })}
-          />
-        </View>
-      )}
+      renderItem={({ item: credential, index }) => {
+        return (
```
Comment (wadeking98) [inline core/App/screens/ListCredentials.tsx:29]: is this change needed? seems like the template is just being wrapped in a `return (...)`. I don't mind either way, just curious

Comment (amanji) [reply core/App/screens/ListCredentials.tsx:29]: Probably just carry over from testing

**Review by wadeking98 — APPROVED** [review]

File: core/App/screens/ProofRequest.tsx (hunk 2 above)
Comment (wadeking98) [reply core/App/screens/ProofRequest.tsx:127]: Just wondering if we're putting too much in one PR

Comment (amanji) [reply core/App/screens/ProofRequest.tsx:127]: Styling revoked credentials are part of the feature. I could leave it out but then wed have to wait on another PR to get this feature fully tested.

**Review by cvarjao — APPROVED** [review]
LGTM
